src/resource.ts

```diff
@@ -54,4 +70,8 @@ export abstract class Resource<T> {
         return this.namePrefix;
     }
 
-}
\ No newline at end of file
+    public getOutputs(): any {
```

Comment:
could this just be `public abstract getOutputs(): any`? Let the implementing class denote if its an empty object or not.

Reply:
Awesome, done as well!